ultravox/data/datasets.py

```diff
@@ -1126,24 +523,26 @@ class InterleaveDataset(SizedIterableDataset):
     def __init__(
         self,
         datasets: Sequence[SizedIterableDataset],
+        weights: Optional[Sequence[float]] = None,
         stop_strategy: StopStrategy = StopStrategy.LAST_EXHAUSTED,
         seed: Optional[int] = 42,
         static: bool = False,
     ) -> None:
         """
         Args:
             datasets: A list of SizedIterableDataset objects.
+            weights: A list of weights for each dataset.
             stop_strategy: Strategy for stopping iteration.
             seed: Optional seed for reproducibility.
             static: If true, the datasets are interleaved in a static order with equal weights.
         """
         self._datasets = datasets
         self._rng = np.random.default_rng(seed)
         self._static = static
-
         self._stop_strategy = stop_strategy
 
-        weights = [getattr(ds, "weight", 1) for ds in datasets]
+        if weights is None:
+            weights = [1.0] * len(datasets)
         total_weight = sum(weights)
```

Comment:
Now that all datasets have a length, so we should update the meaning of weights to represent how many copies (which can be fractional) of each dataset will be used in training. This makes it easier to configure weights in the YAML file. By default, only one copy of each dataset is used for training.

```suggestion
       weights = [w * len(d) for w, d in zip(weights, datasets)]
       total_weight = sum(weights)
```

Reply:
I think this is a more comprehensive change to `InterleavedDataset`, as the `len()` method will need to be adjusted, and the stop strategy changed to run until the overall `len()` is reached. I don't think it's a ton of work but should be done in a separate PR.